app/src/main/kotlin/com/flxrs/dankchat/main/MainFragment.kt

```diff
@@ -403,12 +407,30 @@ class MainFragment : Fragment() {
                         isUserChange = false,
                         loadTwitchData = true,
                     )
-                    val name = dankChatPreferences.userName
-                    if (dankChatPreferences.isLoggedIn && name != null) {
-                        showSnackBar(getString(R.string.snackbar_login, name))
+                    // wait for username to be validated before showing snackbar
+                    collectFlow(dankChatViewModel.currentUserName) {
+                        if (dankChatPreferences.isLoggedIn && it != null) {
+                            showSnackBar(getString(R.string.snackbar_login, it))
+                        }
                     }
                 }
             }
+            // handle oAuth validation error
+            collectFlow(dankChatViewModel.validationError) {
+                when (it) {
+                    OAuthValidationError.OAuthTokenInvalid      -> {
+                        // I'm too donk to create a Fragment or something sorry
```

Comment:
FeelsDankMan fragments would be overkill for a simple error use-case, the DialogBuilder is perfect here
```suggestion
```